src/sql/src/query_model/hir.rs

```diff
@@ -87,7 +103,65 @@ impl FromHir {
                 }
                 select_id
             }
-            _ => panic!("unsupported expression type"),
+            HirRelationExpr::Join {
+                left,
+                right,
+                on,
+                kind,
+            } => {
+                let box_type = if matches!(
+                    kind,
+                    JoinKind::LeftOuter { .. } | JoinKind::RightOuter { .. } | JoinKind::FullOuter
+                ) {
+                    BoxType::OuterJoin(OuterJoin::default())
+                } else {
+                    BoxType::Select(Select::default())
+                };
+                let join_box = self.model.make_box(box_type);
+
+                // Left box
+                let left_box = self.generate_internal(left);
+                let left_q_type = if matches!(kind, JoinKind::LeftOuter { .. }) {
+                    QuantifierType::PreservedForeach
+                } else {
+                    QuantifierType::Foreach
+                };
+
+                let _ = self.model.make_quantifier(left_q_type, left_box, join_box);
+
+                // Right box
+                let right_box = if kind.is_lateral() {
+                    self.within_context(join_box, &mut |generator| -> BoxId {
+                        generator.generate_internal(right)
+                    })
+                } else {
+                    self.generate_internal(right)
+                };
+
+                let right_q_type = if matches!(kind, JoinKind::RightOuter { .. }) {
+                    QuantifierType::PreservedForeach
+                } else {
+                    QuantifierType::Foreach
+                };
+
+                let _ = self
```

Comment:
Will we always ignore the output of this method, or is using it a TODO at the moment?

Reply:
We will always ignore it here, but when going directly from AST to QGM the result returned by `make_quantifier` is actually used.

I'll remove the `let _ =` here.